pkg/session/session.go

```diff
@@ -46,15 +47,17 @@ type Session struct {
 // already exist.
 func GetOrCreate(
 	ctx context.Context,
+	logger logr.Logger,
 	server, datacenter, username, password string, thumbprint string) (*Session, error) {
 
 	sessionMU.Lock()
 	defer sessionMU.Unlock()
 
 	sessionKey := server + username + datacenter
-	if session, ok := sessionCache[sessionKey]; ok {
-		if ok, _ := session.SessionManager.SessionIsActive(ctx); ok {
-			return &session, nil
+	if cachedSession, ok := sessionCache[sessionKey]; ok {
+		if ok, _ := cachedSession.SessionManager.SessionIsActive(ctx); ok {
```

Comment:
Looking into that.

Reply:
I could not repro the session being inactive at my end. I waited for a couple of hours for the invalidation, still couldn't repro it. My vote is with logging the error message as a warning, since I do not think it will contain anything sensitive. Looked through the code, did not find anything obviously sensitive.